lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/ProcessCriCallbackHandler.java

```diff
@@ -145,12 +145,16 @@ public APIGatewayProxyResponseEvent handleRequest(
         } catch (InvalidCriCallbackRequestException e) {
             if (e.getErrorResponse() == ErrorResponse.NO_IPV_FOR_CRI_OAUTH_SESSION) {
                 LOGGER.error(e.getErrorResponse(), e);
+                var pageOutput = StepFunctionHelpers.generatePageOutputMap(
+                        "error",
+                        HttpStatus.SC_UNAUTHORIZED,
+                        PYI_TIMEOUT_RECOVERABLE_PAGE_ID);
+                assert callbackRequest != null;
+                var criOAuthSessionItem = criOAuthSessionService.getCriOauthSessionItem(callbackRequest.getState());
```

Comment:
The only thing that could go wrong is if criOAuthSessionService.getCriOauthSessionItem can't get the session item from the id

Reply:
If you're worried, you could wrap that in a try catch as well, and just swallow the error for now (so at worst it does what it already does)

Reply:
I don't think I'm worried because it's just this:
```
    public CriOAuthSessionItem getCriOauthSessionItem(String criOAuthSessionId) {
        return dataStore.getItem(criOAuthSessionId);
    }
```
so would throw a runtime error which isn't caught anyway (my only concern was now we're running this inside a catch statement, what if we rethrow a caught error)

Reply:
ok sure, if we get a dynamodb error that's not an issue, I guess it could also return null though?

Reply:
I don't think it was caught before